test/test-logger.js

```diff
@@ -27,8 +33,218 @@ console._stdout.write = function() {
   orig.apply(this, arguments);
 };
 
+describe('mocking fd logger with config', function () {
+  it('should write an entry to the fs mock when log is called', function () {
+    var logger = require('../lib/logger.js');
+    var l = logger.create(logger.SILLY, PREFIX, 9);
+    for (var i = 0; i < 15; i += 1) {
+      l.info('info_');
+    }
+    var s = fs.readFileSync(MOCK_PATH).toString();
+    assert.deepEqual(s.split('\n').length, 10, 'There should be 10 entries in the logs')
+  });
+  before(function () {
+    var mockTree = {};
+    mockTree[os.tmpdir()] = {};
+    mockTree[os.tmpdir()][MOCK_FILE] = '';
+    fsMock(mockTree, {createTmp: false});
+  });
+  after(function () {
+    fsMock.restore();
```

Comment:
The `mock-fs` file cache is an in-memory object store that intercepts calls to `fs`. The `restore` function cleans the in-memory store and restores un-intercepted access `fs` while calling the module causes it to reinsert the interceptors and create the in-memory cache again. So no temp files created with these added tests and none that need to be cleaned-up.